src/Build.UnitTests/Evaluation/ProjectSdkImplicitImport_Tests.cs

```diff
@@ -534,6 +542,257 @@ public void IgnoreMissingImportsSdkNotFoundDoesNotThrow()
+                null
+            );
+
+            var projectOptions = SdkUtilities.CreateProjectOptionsWithResolver(
+                new MockExpandedSdkResolver(_testSdkDirectory)
+            );
+
+            void AddProperty(string name, string value) =>
+                (projectOptions.GlobalProperties ??= new Dictionary<string, string>()).Add(name, value);
+
+            if (data.SetNameProperty)
+                AddProperty(SdkNamePropertyName, SdkName);
+
+            if (data.SetVersionProperty)
+                AddProperty(SdkVersionPropertyName, SdkExpectedVersion);
+
+            using var xmlReader = XmlReader.Create(new StringReader(projectContents));
+
+            if (!data.ExpectedSuccess)
+                projectOptions.LoadSettings |= ProjectLoadSettings.IgnoreMissingImports;
+
+            var project = Project.FromXmlReader(xmlReader, projectOptions);
+
+            if (data.ExpectedSuccess)
+            {
+                var expectedSdkPath = Path.GetDirectoryName(_sdkPropsPath);
+
+                // self-consistency check
+                expectedSdkPath.ShouldBe(Path.GetDirectoryName(_sdkTargetsPath));
+
+                var expectedSdkReferenceRaw = new SdkReference(
+                    data.TemplateName,
+                    data.TemplateVersion,
+                    data.Template.ExpectedMinimumVersion
+                );
+
+                var expectedSdkReference = new SdkReference(
+                    SdkName,
+                    SdkExpectedVersion,
+                    data.Template.ExpectedMinimumVersion
+                );
+
+                project.Imports.Count.ShouldBe(2);
+
+                foreach (var import in project.Imports)
+                {
+                    import.ImportingElement.SdkReference.ShouldBe(expectedSdkReferenceRaw);
+                    import.SdkResult.Success.ShouldBeTrue();
+                    import.SdkResult.SdkReference.ShouldBe(expectedSdkReference);
+                    import.SdkResult.Path.ShouldBe(expectedSdkPath);
+                    import.SdkResult.Version.ShouldBe(expectedSdkReference.Version);
+                }
+            }
+            else
+            {
+                project.Imports.Count.ShouldBe(0);
+            }
+        }
+
+        public static IEnumerable<object[]> SdkPropertiesAreExpandedData
+        {
+            get
+            {
+                static IEnumerable<SdkReferencePropertyExpansionMode?> Modes()
+                {
+                    yield return null;
+                    yield return SdkReferencePropertyExpansionMode.NoExpansion;
+                    yield return SdkReferencePropertyExpansionMode.DefaultExpand;
+                    yield return SdkReferencePropertyExpansionMode.ExpandUnescape;
+                    yield return SdkReferencePropertyExpansionMode.ExpandLeaveEscaped;
+                }
+
+                static IEnumerable<(SdkPropertiesAreExpandedDataTemplate, bool, bool)> Templates()
```

Comment:
nit: name the tuple elements to self document the two bools

Reply:
I had them initially documented, but then the hard wrap at 120 characters made the function definition look bad, and I decided since this method is used only once (and can only be used within this property), and the return value is immediately deconstructed with meaningful variable names, this can be left with default tuple field names. I can add the tuple field names, but this'll make method definition look worse. Your call.

Reply:
As middle ground you can only name the two bools (`setName, setVersion`, not that long) and leave the first tuple element unnamed, since the type is expressive already.